src/controllers/SelectYearController.ts

```diff
@@ -0,0 +1,137 @@
+import { APPLICATION_DATA_UNDEFINED, SESSION_NOT_FOUND_ERROR } from 'app/utils/CommonErrors';
+import { PENALTY_DETAILS_PAGE_URI, REVIEW_PENALTY_PAGE_URI, SELECT_THE_PENALTY_PAGE_URI } from 'app/utils/Paths';
+import { ValidationResult } from 'app/utils/validation/ValidationResult';
+
+const template = 'select-the-penalty';
+
+const navigation = {
+    previous(): string {
+        return PENALTY_DETAILS_PAGE_URI;
+    },
+    next(): string {
+        return REVIEW_PENALTY_PAGE_URI;
+    },
+    actions: (_: boolean) => {
+        return {
+            continue:'action=continue'
+        };
+    }
+};
+
+@provide(Processor)
+class Processor implements FormActionProcessor {
+    public process(request: Request): void | Promise<void> {
+
+        const session = request.session;
+
+        if (!session) {
+            throw SESSION_NOT_FOUND_ERROR;
+        }
+
+        const appData: ApplicationData | undefined = session.getExtraData(APPLICATION_DATA_KEY);
+
+        if(!appData) {
+            throw APPLICATION_DATA_UNDEFINED;
+        }
+
+        appData.appeal.penaltyIdentifier.penaltyReference = request.body.selectPenalty;
+
+        session.setExtraData(APPLICATION_DATA_KEY, appData);
+    }
+
+}
+
+// tslint:disable-next-line: max-classes-per-file
+@controller(SELECT_THE_PENALTY_PAGE_URI, SessionMiddleware, AuthMiddleware, PenaltyReferenceRouter)
+export class SelectYearController extends SafeNavigationBaseController<any> {
+
+    public static PENALTY_EXPECTED_ERROR: string = 'Penalty object expected but none found';
+
+    constructor() {
+        super(
+            template,
+            navigation,
+            new FormValidator(schema),
+            undefined,
+            [Processor]
+        );
+    }
+
+    public prepareViewModelFromAppeal(appeal: Appeal): Record<string, any> & any {
+
+        const penaltyList: PenaltyList | undefined = appeal.penaltyIdentifier.penaltyList;
+
+        if (!penaltyList || !penaltyList.items) {
+            throw new Error(SelectYearController.PENALTY_EXPECTED_ERROR);
+        }
+
+        return { penalties: penaltyList.items.map(createPenaltyRadioButton) };
+    }
+
+    protected getExtraActionHandlers(): Record<string, FormActionHandler | FormActionHandlerConstructor> {
+        const that = this;
+        return {
+            continue: {
+                handle: async (request: Request, response: Response) => {
+                    if (that.validator != null) {
+                        const validationResult: ValidationResult = await that.validator.validate(request);
+                        if (validationResult.errors.length > 0) {
+                            return await that.renderWithStatus(UNPROCESSABLE_ENTITY)(
+                                that.template,
+                                {
+                                    ...request.body,
+                                    validationResult,
+                                    ...that.prepareViewModel(),
```

Comment:
I'm wondering if other controllers would benefit from adding this line to the Base Controller's getDefaultActionHandler().. or can we not just add a custom validator that retains this data in some way?

Reply:
It's probably not a safe bet as some keys in the validation result can be the same as in the view model. Custom validator wouldn't really make it simpler as I would have to reconstruct the radio buttons in it

Reply:
If the default action handler was comprised of made of smaller reusable actions *hint* then this wouldn't be a problem

Reply:
I feel it would be cleaner if we created a custom validator over duplicating the default action handler, if it's not too much effort? We can reuse the logic we have in place to construct the radio buttons